src/FieldSet/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { Children, cloneElement } from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+import { statusPropType } from '../common-prop-types'
+import styles from './styles'
+
+const FieldSet = ({
+    className,
+    children,
+    stacked,
+    label,
+    valid,
+    warning,
+    error,
+    required,
+}) => (
+    <fieldset className={cx(className)}>
+        {label && (
+            <legend className={cx({ valid, warning, error, required })}>
+                {label}
+            </legend>
+        )}
+        <div className={cx({ stacked })}>
+            {Children.map(children, child =>
+                cloneElement(child, {
+                    className: cx(child.props.className, 'fieldset-input'),
+                    valid,
```

Comment:
Always forwarding the statuses is not good in my opinion.
A `fieldset` can also be used to group logical inputs that don't have the same `name` attribute, like this:

```jsx
<fieldset>
	<legend>Personal details</legend>
	<label for="first-name">First name</label>
	<input name="first-name" id="first-name" type="text">
	<label for="family-name">Family name</label>
	<input name="family-name" id="family-name" type="text">
	<label for="email">Email</label>
	<input name="email" id="email" type="text">
</fieldset>
```
(source: http://usability.com.au/2013/04/accessible-forms-1-labels-and-identification/)

<hr />

I'd propose a `forwardStatus` prop that defaults to false (or no default, which is falsy)

Reply:
Good that you bring this up, because I was having second thoughts about this too. Upon reflection, I actually don't think we should forward these props at all. By forwarding these props we could be creating unexpected behaviour:
- suppose the `error` prop of the `FieldSet` is set to `true`
- and the `error` prop of a child `Radio` is set to `false`
- because of the way the component prop are merged with the object being passed as a second argument of `cloneElement`, the `Radio` will now have prop `error` set to `true`
- So effectively the parent props are overriding the props set directly on the child. This is not good...

So, I will stop forwarding these props, and in `cloneElement` I will only add the `fieldset-input` to the classNames. This way we have a clearer separation of responsibility.